### PR TITLE
added identity document to get it to work for the Java SDK

### PR DESCRIPTION
**Pull request by jnawk**

### CODE REVIEW

File: metadata/routes.py [hunk 1]
```diff
@@ -69,6 +69,24 @@ def get_credentials():
...
+@route('/latest/dynamic/instance-identity/document')
+def get_identity_document():
+    return {
+        'privateIp' : '127.0.0.1',
+        'devpayProductCodes' : None,
+        'availabilityZone' : 'us-east-1a',
+        'version' : '2010-08-31',
+        'instanceId' : 'i-12345678',
+        'billingProducts' : None,
+        'instanceType' : 't2.nano',
```
Comment (dump247) [inline metadata/routes.py]: The instance type returned by the `/latest/meta-data/instance-type` route above is `r3.2xlarge`. I don't think it makes a difference what the returned instance type is, but it should be consistent. Add a constant to the top of the file (e.g. `INSTANCE_TYPE = 't2.nano'`) and use that in both places.

File: metadata/routes.py [hunk 2]
```diff
@@ -69,6 +69,24 @@ def get_credentials():
...
+        'accountId' : '123456789012',
+        'architecture' : 'x86_64',
+        'kernelId' : None,
+        'ramdiskId' : None,
+        'imageId' : 'ami-12345678',
+        'pendingTime' : '2016-01-01T00:00:00Z',
+        'region' : 'us-east-1'
```
Comment (dump247) [inline metadata/routes.py]: You can get the region from the current profile.

```python
metadata = request.app.config.meta_get('metadata', 'obj')
metadata.profile.region
```

File: metadata/routes.py [hunk 3]
```diff
@@ -69,6 +69,24 @@ def get_credentials():
...
+def get_identity_document():
+    return {
+        'privateIp' : '127.0.0.1',
+        'devpayProductCodes' : None,
+        'availabilityZone' : 'us-east-1a',
+        'version' : '2010-08-31',
+        'instanceId' : 'i-12345678',
+        'billingProducts' : None,
+        'instanceType' : 't2.nano',
+        'accountId' : '123456789012',
```
Comment (dump247) [inline metadata/routes.py]: You could get the account id using the boto session. I remember it was a bit of hassle the last time I had to do it. Something like parsing it of an error response. That was if you wanted to do it consistently, even if the iam account does not have GetUser. Anyway, probably not worth doing it here.

File: metadata/routes.py (hunk 1 above)
Comment (jnawk) [inline metadata/routes.py]: I called it on a t2.nano, ergo it returned t2.nano.  Agree, it doesn't matter, what matters is that this endpoint returns something.

File: metadata/routes.py (hunk 3 above)
Comment (jnawk) [inline metadata/routes.py]: again, doesn't seem to matter.

File: metadata/routes.py (hunk 2 above)
Comment (jnawk) [inline metadata/routes.py]: again, doesn't matter.  you're faking the metadata service, you're not actually in a region anyway.

**Comment by jnawk** [comment]
no changes required.  The Java SDK invokes the endpoint and won't progress if this returns 404.  The actual values returned don't seem to matter; I modelled them on an actual response I got, and changed the values which identify the instance I invoked it on.

I agree some of the values might be provided as parameters, but that can be someone else's PR if they need that.

**Pull request merged**